src/main/java/co/cask/gcs/sink/GCSBatchSink.java

```diff
@@ -80,103 +61,79 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
-    }
-    return GSON.toJson(providedProperties);
-  }
+  protected abstract String getOutputFormatClassname();
+
+  protected abstract Map<String, String> getOutputFormatConfig();
 
   @VisibleForTesting
   GCSBatchSinkConfig getConfig() {
     return config;
   }
 
   /**
-   * S3 Sink configuration.
+   * Sink configuration.
    */
   public static class GCSBatchSinkConfig extends ReferenceConfig {
     @Name("path")
-    @Description("The GCS path where the data is stored. Example: 'gs://logs'.")
+    @Description("The path to write to. For example, gs://<bucket>/path/to/directory")
     @Macro
     protected String path;
 
-    @Name("format")
-    @Description("The format for the path that will be suffixed to the " +
-      "base path; for example: the format 'yyyy-MM-dd-HH-mm' will create a file path ending in '2015-01-01-20-42'. " +
-      "Default format used is 'yyyy-MM-dd-HH-mm'.")
+    @Description("The time format for the output directory that will be appended to the path. " +
+      "For example, the format 'yyyy-MM-dd-HH-mm' will result in a directory of the form '2015-01-01-20-42'. " +
+      "If not specified, nothing will be appended to the path.")
     @Nullable
     @Macro
-    protected String format;
+    protected String suffix;
 
-    @Name("properties")
-    @Description("A JSON string representing a map of properties " +
-      "needed for the distributed file system.")
-    @Nullable
-    @Macro
-    protected String properties;
-
-    @Name("project")
-    @Description("Project ID")
+    @Description(GCPConfig.PROJECT_DESC)
     @Macro
     @Nullable
     protected String project;
 
-    @Name("serviceFilePath")
-    @Description("Service account file path.")
+    @Description(GCPConfig.SERVICE_ACCOUNT_DESC)
     @Macro
     @Nullable
-    protected String serviceAccountFilePath;
+    protected String serviceFilePath;
 
-    @Name("bucket")
     @Description("Name of the bucket.")
     @Macro
     protected String bucket;
 
     public GCSBatchSinkConfig() {
       // Set default value for Nullable properties.
       super("");
-      this.format = DEFAULT_PATH_FORMAT;
-      this.properties = updateFileSystemProperties(path, null);
-    }
-
-    public GCSBatchSinkConfig(String referenceName, String path,
-                              @Nullable String format, @Nullable String properties,
-                              String project, String serviceAccountFilePath, String bucket) {
-      super(referenceName);
-      this.path = path;
-      this.format = format == null || format.isEmpty() ? DEFAULT_PATH_FORMAT : format;
-      this.project = project;
-      this.serviceAccountFilePath = serviceAccountFilePath;
-      this.bucket = bucket;
-      this.properties = updateFileSystemProperties(path, properties);
     }
 
     public void validate() {
+      // validate project id availability
+      GCPUtil.getProjectId(project);
       if (path != null && !path.startsWith("gs://")) {
```

Comment:
should check projectId, path are not a macro.

Reply:
done